common/src/main/java/com/wynntils/features/user/ChatShowRealNameFeature.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.handlers.chat.event.ChatMessageReceivedEvent;
+import java.util.List;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.HoverEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ChatShowRealNameFeature extends UserFeature {
+    // credits to avomod for the nickname detection method
+    private static Component eventOriginalMessage;
+    private static String eventOriginalColorMessage;
+
+    @SubscribeEvent
+    public void onChatMessage(ChatMessageReceivedEvent event) {
+        eventOriginalMessage = event.getOriginalMessage();
+        eventOriginalColorMessage = event.getOriginalCodedMessage();
+        addRealNameToMessage(event.getOriginalMessage(), event.getOriginalMessage());
```

Comment:
Why do you pass the argument twice?

Reply:
Oh, for some reason merging overwrote my changes, I'll see what else got overwritten.

Reply:
Nevermind, this is intended as the method is recursive and will use other values for the arguments later on.

Reply:
Why do you make it recursive?! :-o We try to use an iterative style in the mod; java is not really suitable for recursive programming.

Reply:
This is because the part of the message with the name and nickname is nested deep within the message and is in different locations for different messages (for example guild messages/public chat/private messages etc)